### PR TITLE
Set a specific initrd path for c2, in case the kernel is the same.

### PR DESCRIPTION
**Pull request by mherrb**
Otherwise this script will overwrite the initrd used by c1,
causing havoc later.

Signed-off-by: Matthieu Herrb <matthieu.herrb@laas.fr>

### CODE REVIEW

**Comment by mherrb** [comment]
This one is an easy one from my local changes.

**Comment by mherrb** [comment]
hmm the "don't" re-export /home on c2 commit above was not supposed to be added to this pull request. It's a separate (small) issue.

**Comment by TheDash** [comment]
Thanks for the PR.

Does C2 find the .c2 initrd without issue? My understanding is that it serves the initrd from the pxelinux.cfg directory and there is a symlink which points to the initrd. Does anything have to be updated there?

**Comment by mherrb** [comment]
Ok So everything I push to my cloned repository ends up in a single pull request. I learned something about Github today…

To answer your question, yes. the netboot-kernel-links scripts correctly sets the link in /var/lib/tftpboot/robot64/ to the .c2 initrd file, so no problem to find it.

File: root/var/lib/tftpboot/pxelinux.cfg/0A440002 [hunk 1]
```diff
@@ -3,7 +3,7 @@ DEFAULT robot64
 TIMEOUT 3
 LABEL robot64
         kernel robot64/vmlinuz
-        append boot=nfs root=/dev/nfs selinux=0 initrd=robot64/initrd.img rw rootdelay=10 acpi=off numa=noacpi vga=extended nfsroot=10.68.0.1:/ ip=10.68.0.2:10.68.0.1:10.68.0.1:255.255.255.0:c2:eth1 noefi nomodeset init=/usr/sbin/c2init console=ttyS0,57600 console=tty1 biosdevname=0 net.ifnames=1
+        append boot=nfs root=/dev/nfs selinux=0 initrd=robot64/initrd.img rw rootdelay=10 vga=extended nfsroot=10.68.0.1:/ ip=10.68.0.2:10.68.0.1:10.68.0.1:255.255.255.0:c2:lan0 nomodeset init=/usr/sbin/c2init console=ttyS0,57600 console=tty1 biosdevname=0 net.ifnames=1
```
Comment (TheDash) [inline root/var/lib/tftpboot/pxelinux.cfg/0A440002]: Are you sure this is the working configuration? I'm hesitant to change the lan0 and the udev rules because I've had it be lan0 before, and it has borked the entire system. The units we're shipping now and the images that are cloned for new users, the existing configuration works without problem.

I have gone down the same road that your lab has with changing it and it appearing to work, but it may not be a general enough solution.

The only different thing in the udev rules is the 5 and 6 have swapped. We would need to confirm that this is 100% how the network hardware is setting these ports. At the moment, eth1 is picked because it is not renamed, and is static with the physical layout of the network ports on the board.

File: root/unionfs/overlay/etc/udev/rules.d/71-pr2-network-interface.rules [hunk 2]
```diff
@@ -23,5 +23,5 @@ SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="00:1e:68:??:??:?
 SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="00:1e:68:??:??:?[13579bdf]", ATTR{type}=="1", KERNEL=="eth*", NAME="lan1"
 
 # CLEARPATH LAN PORTS
-#SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="d0:50:99:??:??:[02468ace]?", ATTR{type}=="1", NAME="lan0"
-#SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="d0:50:99:??:??:[13579bdf]?", ATTR{type}=="1", NAME="lan1"
+SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="d0:50:99:??:??:?[02458ace]", ATTR{type}=="1", NAME="lan0"
```
Comment (TheDash) [inline root/unionfs/overlay/etc/udev/rules.d/71-pr2-network-interface.rules]: To add; we must be certain about the mac address attribute naming for lan0, lan1, etc before changing them as it has serious ramifactions for all users who do updates and installs. 

I'm going to ask it be put into a separate pull request, and that we tackle only the .c2 naming in this one.

File: root/var/lib/tftpboot/pxelinux.cfg/0A440002 (hunk 1 above)
Comment (mherrb) [inline root/var/lib/tftpboot/pxelinux.cfg/0A440002]: On Wed, May 25, 2016 at 10:10:48AM -0700, Devon Ash wrote:
> > @@ -3,7 +3,7 @@ DEFAULT robot64
> >  TIMEOUT 3
> >  LABEL robot64
> >          kernel robot64/vmlinuz
> > -        append boot=nfs root=/dev/nfs selinux=0 initrd=robot64/initrd.img rw rootdelay=10 acpi=off numa=noacpi vga=extended nfsroot=10.68.0.1:/ ip=10.68.0.2:10.68.0.1:10.68.0.1:255.255.255.0:c2:eth1 noefi nomodeset init=/usr/sbin/c2init console=ttyS0,57600 console=tty1 biosdevname=0 net.ifnames=1
> > +        append boot=nfs root=/dev/nfs selinux=0 initrd=robot64/initrd.img rw rootdelay=10 vga=extended nfsroot=10.68.0.1:/ ip=10.68.0.2:10.68.0.1:10.68.0.1:255.255.255.0:c2:lan0 nomodeset init=/usr/sbin/c2init console=ttyS0,57600 console=tty1 biosdevname=0 net.ifnames=1
> 
> Are you sure this is the working configuration? I'm hesitant to
> change the lan0 and the udev rules because I've had it be lan0
> before, and it has borked the entire system. The units we're
> shipping now and the images that are cloned for new users, the
> existing configuration works without problem.

Yes I'm sure it's the working configuration on our robot with the new
PC. I get lan0 and lan1 as the interface names in this configuration.

I just need to recheck if I had to also do something for the
initrd. I'm not near the robot right now and won't be able to check
before friday.

> 
> I have gone down the same road that your lab has with changing it and it appearing to work, but it may not be a general enough solution.
> 
> The only different thing in the udev rules is the 5 and 6 have
> swapped. We would need to confirm that this is 100% how the network
> hardware is setting these ports. At the moment, eth1 is picked
> because it is not renamed, and is static with the physical layout of
> the network ports on the board.

Unless you can be sure that themotherboard manufacter is following
some logic pattern for the on-board interfaces, I'm not sure you can
have a static udev rules file that will match all configurations.

As I wrote in the commit message, it's probably easier to keep the
eth0/eth1 naming that we get by removing the udev rules alltogether
and just using biosdevname=0.


> 
> ---
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub:
> https://github.com/pr2-debs/pr2-netboot/pull/12/files/051a8ddb724d883b7950853f107e2b51e76cd65c#r64611633

-- 
Matthieu Herrb

File: root/unionfs/overlay/etc/udev/rules.d/71-pr2-network-interface.rules (hunk 2 above)
Comment (mherrb) [inline root/unionfs/overlay/etc/udev/rules.d/71-pr2-network-interface.rules]: On Wed, May 25, 2016 at 10:11:27AM -0700, Devon Ash wrote:
> > @@ -23,5 +23,5 @@ SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="00:1e:68:??:??:?
> >  SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="00:1e:68:??:??:?[13579bdf]", ATTR{type}=="1", KERNEL=="eth*", NAME="lan1"
> >  
> >  # CLEARPATH LAN PORTS
> > -#SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="d0:50:99:??:??:[02468ace]?", ATTR{type}=="1", NAME="lan0"
> > -#SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="d0:50:99:??:??:[13579bdf]?", ATTR{type}=="1", NAME="lan1"
> > +SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="d0:50:99:??:??:?[02458ace]", ATTR{type}=="1", NAME="lan0"
> 
> To add; we must be certain about the mac address attribute naming for lan0, lan1, etc before changing them as it has serious ramifactions for all users who do updates and installs. 
> 
> I'm going to ask it be put into a separate pull request, and that we
> tackle only the .c2 naming in this one.

Yes sure. I've been surprised that github adds all subsequents commits
to my fork of the repo to the existing pull request, but I've been
lazy to create one branch per patch to avoid this.

I'll re-submmit separate pull requests tomorrow.

> 
> ---
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub:
> https://github.com/pr2-debs/pr2-netboot/pull/12/files/051a8ddb724d883b7950853f107e2b51e76cd65c#r64611759

-- 
Matthieu Herrb

File: root/var/lib/tftpboot/pxelinux.cfg/0A440002 (hunk 1 above)
Comment (TheDash) [inline root/var/lib/tftpboot/pxelinux.cfg/0A440002]: > I just need to recheck if I had to also do something for the
> initrd. I'm not near the robot right now and won't be able to check
> before friday.

We changed the initrd to boot off of eth1 as well, inside the initramfs.conf script. It used to be lan1. 

> Unless you can be sure that themotherboard manufacter is following
> some logic pattern for the on-board interfaces, I'm not sure you can
> have a static udev rules file that will match all configurations.

Which is why the lan0 and lan1 renaming scheme with udev rules is not used here. Naming via udev may work for your machine, but not all of our machines. There is a renaming scheme the kernel uses called biosdevname which picks ethX and names them by physical layout, which is what we use now.

> As I wrote in the commit message, it's probably easier to keep the
> eth0/eth1 naming that we get by removing the udev rules alltogether
> and just using biosdevname=0.

Ah yes, we're on the same page.

**Comment by mherrb** [comment]
I'm closing this PR and will be creating new separate ones.
